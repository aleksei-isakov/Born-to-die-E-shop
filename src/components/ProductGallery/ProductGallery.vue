<template>
  <div class="product-gallery__wrapper">
    <Slider :images="images" />
    <AddToCartBlock
      :is-sign-in="false"
      :profile="mockprofile"
      :rating="rating"
      class="add-to-cart"
    />
  </div>
</template>

<script>
import Slider from '@/components/Slider/Slider';
import AddToCartBlock from '@/components/Slider/AddToCartBlock/AddToCartBlock';
import mockprofile from '@/components/Slider/AddToCartBlock/mockprofile.json';

export default {
  name: 'ProductGallery',

  components: { AddToCartBlock, Slider },

  props: {
    images: {
      type: Array,
      required: true,
      default: () => []
    },

    rating: {
      type: Number,
      default: 0,
      required: true
    }
  },

  data() {
    return {
      mockprofile: mockprofile
    };
  }
};
</script>

<style lang="scss" scoped>
@import '@/scss/CustomVariables.scss';

.product-gallery__wrapper {
  display: flex;
}

@media screen and (max-width: $tablet-size) {
  .product-gallery__wrapper {
    flex-direction: column;
  }
}
</style>
