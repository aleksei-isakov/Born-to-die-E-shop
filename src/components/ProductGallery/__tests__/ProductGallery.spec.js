import { shallowMount } from '@vue/test-utils';
import ProductGallery from '../ProductGallery';

let wrapper;

beforeAll(() => {
  wrapper = shallowMount(ProductGallery, {
    propsData: {
      images: [
        'http://placeimg.com/640/480',
        'http://placeimg.com/640/480',
        'http://placeimg.com/640/480'
      ]
    }
  });
});
afterAll(() => {
  wrapper.destroy();
});
describe('ProductGallery', () => {
  test('should contain Slider and AddToCartBlock components', () => {
    expect(wrapper.find('Slider', 'AddToCartBlock')).toBeTruthy();
  });
  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
