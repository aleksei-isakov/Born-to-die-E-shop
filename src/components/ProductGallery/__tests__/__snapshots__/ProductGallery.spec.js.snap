// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductGallery should match snapshot 1`] = `
<div class="product-gallery__wrapper">
  <slider-stub images="http://placeimg.com/640/480,http://placeimg.com/640/480,http://placeimg.com/640/480"></slider-stub>
  <addtocartblock-stub profile="[object Object]" rating="0" class="add-to-cart"></addtocartblock-stub>
</div>
`;
